src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -175,7 +172,11 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
             
             if (triggerAction == TDTM_Runnable.Action.AfterInsert)
                 copyRecurringDonationCampaignAllocations(listOppsForProcessing);
-            
+
```

Comment:
Yeah, that sounds like a bug and we should prefer the campaign allocations. Please do create an issue for it.